### PR TITLE
Do not give everyone permission to read answers

### PR DESCRIPTION
**Pull request by Natim**
Refs https://github.com/Kinto/kinto/pull/828#issuecomment-249881244
r? @almet 
f? @enguerran 

When we created the formbuilder it wasn't possible to let people read the schema without being able to read all the form answer.

This limitation was fixed in Kinto 4.3 and people with the `record:create` permission can now read the collection schema and their answer only.

This is the implementation of the fix for the formbuilder.

### CODE REVIEW

File: formbuilder/actions/server.js [hunk 1]
```diff
@@ -167,14 +154,14 @@ export function submitRecord(record, collection, callback) {
   };
 }
 
-export function loadSchema(collection, callback) {
+export function loadSchema(formID, callback) {
   return (dispatch, getState) => {
     dispatch({type: SCHEMA_RETRIEVAL_PENDING});
     new KintoClient(config.server.remote, {
-      headers: getAuthenticationHeaders(collection)
+      headers: getAuthenticationHeaders("EVERYONE")
```
Comment (Natim) [inline formbuilder/actions/server.js]: Could be "AUTHENTICATED" or "ANYONE"

File: formbuilder/utils.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import config from "./config";
  * are stored. This is useful to always have one id to pass to the clients,
  * and they can figure out what the user token and collection name is.
  **/
-export function getUserToken(adminToken) {
+export function getFormID(adminToken) {
```
Comment (almet) [inline formbuilder/utils.js]: If you change the name of the function, you need to update the description for it as well (returns the unique identifier of the form, from the administrative token)

**Comment by Natim** [comment]
Note that this needs Kinto >= 4.3

**Comment by enguerran** [comment]
I test the new formbuilder with the https://kinto.dev.mozaws.net/v1/ backend.

The admin token is: 91ec309e9a3c463ca63fd8036d9e6d71
The form ID is: 91ec309e9a3c463c

As you can see in the [admin console](https://kinto.github.io/kinto-admin/#/buckets/formbuilder/collections/91ec309e9a3c463c/records?_k=klsxw0) it looks ok! 👍 

I suggest to update the README.md as it still refers to `userToken` instead of `formID`.

I find a bug too when you create a form without any required field, it fails publishing the form → https://github.com/Kinto/formbuilder/pull/156/commits/2502118ab8ab2edccbeda6cdd9389ff31de7e2dd#diff-d3be7b791c4d0d2548517ed3b5b9834bR81

**Comment by Natim** [comment]
I removed the PUBLICATION_FAILED so I should probably put it back.

**Comment by enguerran** [comment]
In fact, I cannot reproduce. I got an error in the console
```
TypeError: schema.required is undefined
```

**Comment by Natim** [comment]
Yes but the TypeError should change the form status as well.

**Comment by enguerran** [comment]
You can add something like that to forbid empty form publication:

```
    if(Object.keys(schema.properties).length === 0) {
      dispatch(addNotification("We cannot publish an empty form.", {type: "error"}));
      dispatch({ type: FORM_PUBLICATION_FAILED });
    }
```

**Comment by almet** [comment]
We should already have this somewhere. It rings a bell :)

**Comment by almet** [comment]
@natim, how does this affects people with already in use forms? Does this continues to work for them or is this breaking compatibility? Thanks for the precision, and thanks for the code submission :-)

**Comment by almet** [comment]
This is now ready to be merged!

**Pull request merged**